File: schemas/e-commerce/drizzle/wishlists.ts
// wishlists: Named, optionally public product wishlists owned by a user.
// See README.md for full design rationale.

import { pgTable, uuid, text, boolean, timestamp, index } from "drizzle-orm/pg-core";
import { users } from "../../auth-rbac/drizzle/users";

export const wishlists = pgTable(
  "wishlists",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    userId: uuid("user_id")
      .notNull()
      .references(() => users.id, { onDelete: "cascade" }),
    name: text("name").notNull().default("Default"),
    isPublic: boolean("is_public").notNull().default(false),
    createdAt: timestamp("created_at", { withTimezone: true })
      .notNull()
      .defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true })
      .notNull()
      .defaultNow()
      .$onUpdateFn(() => new Date()),
  },
  (table) => [index("idx_wishlists_user_id").on(table.userId)]
);

File: schemas/e-commerce/spacetimedb/wishlists.rs
// wishlists: User-curated lists of saved products for future purchase.
// See README.md for full design rationale.
use spacetimedb::Timestamp;

#[spacetimedb::table(name = wishlists, public)]
pub struct Wishlist {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub user_id: String, // FK → users.id (cascade delete)
    pub name: String,
    pub is_public: bool,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
